Analytics/Analytics/Base/AnalyticsClient.swift

```diff
@@ -35,20 +35,60 @@ public class AnalyticsClient {
      */
     private var processEventChannel: AsyncChannel<Event>
 
+    /**
+     The session manager responsible for handling session operations.
+     */
+    internal var sessionManager: SessionManager
+    
     /**
      Initializes the `AnalyticsClient` with the given configuration.
      
      - Parameter configuration: The configuration object containing settings and storage details.
      */
     public init(configuration: Configuration) {
         self.configuration = configuration
+        self.sessionManager = SessionManager(storage: configuration.storage)
         self.processEventChannel = AsyncChannel(capacity: Int.max)
         self.userIdentityState = createState(initialState: UserIdentity.initializeState(configuration.storage))
         
         self.setup()
     }
 }
 
+// MARK: - Session
+
+extension AnalyticsClient {
+    
+    /**
+     Starts a session with a given `sessionId`, or generates one if not provided.
+     
+     - Parameter sessionId: An optional `UInt64` session ID. If `nil`, a new session ID is generated.
+     */
+    public func stateSession(sessionId: UInt64? = nil) {
```

Comment:
Why did we keep this type to be of `UInt64`? It should be of type `long` to have parity between Android and iOS!

Reply:
API name should be `startSession`:

```suggestion
    public func startSession(sessionId: UInt64? = nil) {
```

Reply:
> Why did we keep this type to be of `UInt64`? It should be of type `long` to have parity between Android and iOS!

- There is no `long` in Swift. The equivalent for `long` is `Int64`. I don't want to receive negative values from the user. So it is `UInt64` (Unsigned Integer)

Reply:
Please highlight this fact in the PR description for future reference and to help the PR reviewer.

NOTE: If there is any known disparity, it's always better to highlight the fact in the PR description.

Also, correct the PR description related to the `stateSession`.